src/pytorch_lightning/strategies/strategy.py

```diff
@@ -74,7 +74,10 @@ def accelerator(self, accelerator: "pl.accelerators.accelerator.Accelerator") ->
 
     @property
     def checkpoint_io(self) -> CheckpointIO:
-        return self._checkpoint_io if self._checkpoint_io is not None else TorchCheckpointIO()
+        if self._checkpoint_io is None:
+            self._checkpoint_io = TorchCheckpointIO()
```

Comment:
```suggestion
            self._checkpoint_io: Optional[CheckpointIO] = TorchCheckpointIO()
```

This is from where mypy is getting the types that are throwing the errors. I fyou type this assignment, it works without errors. However, I think it might be nicer to have this variable typed already in `__init__`

Reply:
it's typed in `__init__`, not sure why it's raising an error here

Reply:
That is not true. What is typed in `__init__` is `self.checkpoint_io`, not `self._checkpoint_io`. That is then assigned through setter and I guess mypy does not follow through the setter

Reply:
the other option is to assign to `self._checkpoint_io` directly in `__init__` and not through the setter

Reply:
ah! I thought it was `self._checkpoint_io: Optional[CheckpointIO]` all the time.

thank you :)